mmv1/products/developerconnect/Connection.yaml

```diff
@@ -99,6 +110,7 @@ properties:
           Represents an OAuth token of the account that authorized the Connection,and
           associated metadata.
         min_version: 'beta'
+        default_from_api: true
```

Comment:
My only concern is the inclusion of `default_from_api: true` on these two properties. What is the reason for this?

Reply:
Sure, this requires some product knowledge, but let me try to explain. `default_from_api: true` means if the user leave it unspecified, Terraforms defers to the server provided value instead of sending an empty value that wipes out the server side value. Consider a typical config to set up a connection

```terraform
resource "google_developer_connect_connection" "default" {
  ...

  github_config {
    github_app = "FIREBASE"
    # Note the unspecified authorizer_credentials field
  }
}

output "next_steps" {
  description = "Follow the action_uri if present to continue setup"
  value = google_developer_connect_connection.default.installation_state
}
```

1. Run `terraform apply`, and the output will provide a URL for the user to visit for an OAuth flow
2. After the OAuth flow, `authorizer_credentials` is populated on the server side, but the Terraform configuration doesn't know about it
3. If the user runs `terraform apply` again on the same config, without `default_from_api`, terraform will detect a diff and accidentally wipe out the `authorizer_credentials`
4. The correct behavior is that terraform will refresh to capture the `authorizer_credentials` populated outside of Terraform, and doesn't complain about a diff.
5. If the user wants to use another `authorizer_credentials`, they can do so and Terraform will send the new value to the server

Reply:
Thanks for the explanation, this is all I needed!